stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -1197,23 +1227,50 @@ public class MessageListViewModel(
     }
 
     /**
-     * Scrolls to message if in list otherwise get the message from backend.
+     * Scrolls to message if in list otherwise get the message from backend. Does not work for threads.
      *
      * @param message The message we wish to scroll to.
      */
     public fun scrollToSelectedMessage(message: Message) {
+        if (isInThread) return
+
         val isMessageInList = currentMessagesState.messageItems.firstOrNull {
             it is MessageItemState && it.message.id == message.id
         } != null
+        scrollToMessage = message
 
         if (isMessageInList) {
             focusMessage(message.id)
         } else {
-            scrollToMessage = message
             loadMessage(message = message)
         }
     }
 
+    /**
+     * Requests that the list scrolls to the bottom to the newest messages. If the newest messages are loaded will set
+     * scroll the list to the bottom. If they are not loaded will request the newest data and once loaded will scroll
+     * to the bottom of the list.
+     *
+     * @param messageLimit The message count we wish to load from the API when loading new messages.
+     * @param scrollToBottom Notifies the ui to scroll to the bottom if the newest messages are in the list or have been
+     * loaded from the API.
+     */
+    public fun scrollToBottom(messageLimit: Int = DefaultMessageLimit, scrollToBottom: () -> Unit) {
+        if (isInThread) {
+            scrollToBottom()
+        } else {
+            if (currentMessagesState.startOfMessages) {
+                scrollToBottom()
+                return
+            }
+            chatClient.loadNewestMessages(channelId, messageLimit).enqueue {
```

Comment:
Do we have a guarantee the messages are not loaded already?

Reply:
yes we do, but the function is not implemented properly, might have left it like this after some testing. This should be inside else since the check above checks whether the newest messages are loaded or not. Will fix this.